tests/pubsub/testgossipmembership.nim

```diff
@@ -0,0 +1,255 @@
+  discard
+
+const MsgIdSuccess = "msg id gen success"
+let DURATION_TIMEOUT = 500.milliseconds
+
+suite "GossipSub Topic Membership Tests":
+  teardown:
+    checkTrackers()
+  # Addition of Designed Test cases for 6. Topic Membership Tests: https://www.notion.so/Gossipsub-651e02d4d7894bb2ac1e4edb55f3192d
+
+  # Test for subscribing to a topic and verifying mesh and gossipsub structures
+  asyncTest "handle SUBSCRIBE to the topic":
+    # Given 5 gossipsub nodes
+    let
+      numberOfNodes = 5
+      topic = "test-topic"
+      nodes = generateNodes(numberOfNodes, gossip = true)
+
+    await allFuturesThrowing(nodes.mapIt(it.switch.start()))
+
+    # When all of them are connected and subscribed to the same topic
+    await subscribeNodes(nodes)
+    for node in nodes:
+      node.subscribe(topic, voidTopicHandler)
+
+    await sleepAsync(2 * DURATION_TIMEOUT)
+
+    # Then their related attributes should reflect that
+    for node in nodes:
+      let currentGossip = GossipSub(node)
+      check currentGossip.topics.contains(topic)
+      check currentGossip.gossipsub[topic].len() == numberOfNodes - 1
+      check currentGossip.mesh[topic].len() == numberOfNodes - 1
+
+    await allFuturesThrowing(nodes.mapIt(allFutures(it.switch.stop())))
+
+  # Test unsubscribing from a topic and verifying removal from relevant data structures
+  asyncTest "handle UNSUBSCRIBE to the topic":
+    # Given 5 nodes subscribed to a topic
+    let
+      numberOfNodes = 5
+      topic = "test-topic"
+      nodes = generateNodes(numberOfNodes, gossip = true)
+
+    await allFuturesThrowing(nodes.mapIt(it.switch.start()))
+
+    await subscribeNodes(nodes)
+    for node in nodes:
+      node.subscribe(topic, voidTopicHandler)
+
+    await sleepAsync(2 * DURATION_TIMEOUT)
+
+    # When all nodes unsubscribe from the topic
+    for node in nodes:
+      node.unsubscribe(topic, voidTopicHandler)
+
+    await sleepAsync(2 * DURATION_TIMEOUT)
+
+    # Then the topic should be removed from relevant data structures
+    for node in nodes:
+      let currentGossip = GossipSub(node)
+      check topic notin currentGossip.topics
+      if topic in currentGossip.mesh:
+        check currentGossip.mesh[topic].len == 0
+      else:
+        check topic notin currentGossip.mesh
+      if topic in currentGossip.gossipsub:
+        check currentGossip.gossipsub[topic].len == 0
+      else:
+        check topic notin currentGossip.gossipsub
+
+    await allFuturesThrowing(nodes.mapIt(allFutures(it.switch.stop())))
+
+  # Test subscribing and unsubscribing multiple topics
+  asyncTest "handle SUBSCRIBE and UNSUBSCRIBE multiple topics":
+    # Given 3 nodes and multiple topics
+    let
+      numberOfNodes = 3
+      topics = ["topic1", "topic2", "topic3"].toSeq()
+      nodes = generateNodes(numberOfNodes, gossip = true)
+
+    await allFuturesThrowing(nodes.mapIt(it.switch.start()))
+
+    # When nodes subscribe and then unsubscribe to multiple topics
```

Comment:
I'd remove the second part here. Leave only: `When nodes subscribe to multiple topics`, as that's what you're doing here, and later you're handling the `unsubscribe` part.